src/index.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+var Image = require('image-js');
+
+const defaultOptions = {
+    lowThreshold: 10,
+    highThreshold: 30,
+    blur: 1.1,
+    brightness: 255
+};
+
+
+/**
+ * Find edges in an image using the Canny algorithm
+ * @param {Image} image
+ * @param {object} options
+ * @param {number} [options.lowThreshold=10] : first threshold for the hysteresis procedure.
+ * @param {number} [options.highThreshold=30] : second threshold for the hysteresis procedure.
+ * @param {number} [options.blur=1.1] : sigma parameter for gaussian filter step.
+ * @param {number} [options.brightness] : values assigned to each edge pixel on the result image.
+ * @return {Image} : an image with the edges at options.brightness value.
+ */
+function cannyEdgeDetector(image, options) {
+    options = Object.assign({}, defaultOptions, options);
+
+    var width = image.width, height = image.height;
+
+    image = image.grey({
+        algorithm: 'luma601'
+    });
+
+
+    var Gx = [
+        [-1, 0, +1],
+        [-2, 0, +2],
+        [-1, 0, +1]
+    ];
+
+    var Gy = [
+        [-1, -2, -1],
+        [0, 0, 0],
+        [+1, +2, +1]
+    ];
+
+    var gfOptions = {
+        sigma: options.blur,
+        radius: 3
+    };
+
+    var gf = image.gaussianFilter(gfOptions);
+
+    var convOptions = {
+        bitDepth: 32,
+        mode: 'periodic'
+    };
+
+    var gradientX = gf.convolution(Gy, convOptions);
+    var gradientY = gf.convolution(Gx, convOptions);
+
+    var G = new Image(width, height, {
```

Comment:
all of this until line 71 can be replaced by `var G = gradientY.hypotenuse(gradientX)`